crates/oxc_ast/src/ast/ts.rs

```diff
@@ -931,27 +953,14 @@ pub enum TSModuleDeclarationBody<'a> {
     TSModuleBlock(Box<'a, TSModuleBlock<'a>>),
 }
 
-#[visited_node(scope(ScopeFlags::TsModuleBlock))]
-#[derive(Debug)]
+#[visited_node]
+#[derive(Debug, Hash)]
 #[cfg_attr(feature = "serialize", derive(Serialize, Tsify))]
 #[cfg_attr(feature = "serialize", serde(tag = "type", rename_all = "camelCase"))]
 pub struct TSModuleBlock<'a> {
     #[cfg_attr(feature = "serialize", serde(flatten))]
     pub span: Span,
     pub body: Vec<'a, Statement<'a>>,
-    pub scope_id: Cell<Option<ScopeId>>,
-}
-
-impl<'a> TSModuleBlock<'a> {
-    pub fn new(span: Span, body: Vec<'a, Statement<'a>>) -> Self {
```

Comment:
I think we can keep this method

Reply:
I see. Actually, `PrivateIdentifier` and `IdentifierName` both have new methods, but they don't have `scope_id`. It doesn't matter. I just to say. We can add a `new` method for all AST nodes

Reply:
Ah right, so there is no pattern! OK, I propose to leave this as is for now. As you say, we can add `new` methods for all AST node types later on if we think that'd be of benefit.